runtime/io/common/src/aws/smithy/kotlin/runtime/io/SdkByteReadChannel.kt

```diff
@@ -97,3 +100,34 @@ public suspend fun SdkByteReadChannel.readToBuffer(): SdkBuffer {
     readRemaining(buffer)
     return buffer
 }
+
+/**
+ * Read all bytes from this channel into [sink]. Returns the total number of bytes written.
+ */
+public suspend fun SdkByteReadChannel.readAll(sink: SdkSink): Long = withContext(SdkDispatchers.IO) {
+    val bufferedSink = if (sink is SdkBuffer) sink else sink.buffer()
```

Comment:
**Question**: Doesn't `SdkSink.buffer()` already no-op when `this` is buffered?

Reply:
Yes, probably wasn't true at the time this was added (or was missed in rebasing). I can remove it